### PR TITLE
[infra] Make travis deal with adding a new project or removing one

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This change causes travis to skip building a project when a project does not have build.sh file, a common scenario when a project is added (and only a project.yaml file is added) or removed.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
Fixes #2554

File: infra/travis/travis_build.py [hunk 1]
```diff
@@ -28,14 +28,27 @@
...
+  build.sh file."""
   master_head_sha = subprocess.check_output(
       ['git', 'merge-base', 'HEAD', 'FETCH_HEAD']).decode().strip()
   output = subprocess.check_output(
       ['git', 'diff', '--name-only', 'HEAD', master_head_sha]).decode()
   projects_regex = '.*projects/(?P<name>.*)/.*\n'
-  return set(re.findall(projects_regex, output))
+  modified_projects = set(re.findall(projects_regex, output))
+  projects_dir = os.path.abspath(
+      os.path.join(__file__, '..', '..', '..', 'projects'))
```
Comment (ammaraskar) [inline infra/travis/travis_build.py]: `os.path.join(get_oss_fuzz_root(), 'projects')`?

Comment (inferno-chromium) [reply infra/travis/travis_build.py]: +1

Comment (jonathanmetzman) [reply infra/travis/travis_build.py]: Good catch. Done!

File: infra/travis/travis_build.py [hunk 2]
```diff
@@ -28,14 +28,27 @@
...
+  # Filter out projects without build.sh files since new projects and reverted
+  # projects frequently don't have them. In these cases we don't want Travis's
+  # builds to fail.
+  modified_buildable_projects = []
+  for project in modified_projects:
+    if not os.path.exists(os.path.join(projects_dir, project, 'build.sh')):
+      print('{0} does not have a build.sh. Not building.'.format(project))
```
Comment (inferno-chromium) [inline infra/travis/travis_build.py]: nit: s/Not building/Skipping build ?

Comment (jonathanmetzman) [reply infra/travis/travis_build.py]: Done.

**Pull request merged**